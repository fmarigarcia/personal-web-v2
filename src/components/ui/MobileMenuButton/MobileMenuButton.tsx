import React from 'react';

interface MobileMenuButtonProps {
    onClick?: () => void;
    isOpen?: boolean;
    className?: string;
}

export const MobileMenuButton: React.FC<MobileMenuButtonProps> = ({
    onClick,
    isOpen = false,
    className = '',
}) => {
    return (
        <button
            onClick={onClick}
            className={`md:hidden p-2 text-gray-700 hover:text-blue-600 transition-colors ${className}`}
            type="button"
            aria-label={isOpen ? 'Close mobile menu' : 'Open mobile menu'}
            aria-expanded={isOpen}
        >
            <svg
                className="w-6 h-6"
                fill="none"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                stroke="currentColor"
            >
                {isOpen ? (
                    <path d="M6 18L18 6M6 6l12 12" />
                ) : (
                    <path d="M4 6h16M4 12h16M4 18h16" />
                )}
            </svg>
        </button>
    );
};

export default MobileMenuButton;
