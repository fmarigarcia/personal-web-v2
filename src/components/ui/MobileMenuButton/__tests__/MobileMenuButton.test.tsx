import { render, screen, fireEvent } from '@testing-library/react';
import { MobileMenuButton } from '../MobileMenuButton';

describe('MobileMenuButton', () => {
    it('should render mobile menu button', () => {
        render(<MobileMenuButton />);

        const button = screen.getByRole('button', {
            name: /open mobile menu/i,
        });
        expect(button).toBeInTheDocument();
    });

    it('should handle click when onClick prop is provided', () => {
        const mockOnClick = jest.fn();
        render(<MobileMenuButton onClick={mockOnClick} />);

        const button = screen.getByRole('button');
        fireEvent.click(button);

        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });

    it('should show hamburger icon when menu is closed', () => {
        render(<MobileMenuButton isOpen={false} />);

        const button = screen.getByRole('button');
        const svg = button.querySelector('svg');
        const path = svg?.querySelector('path');

        expect(path).toHaveAttribute('d', 'M4 6h16M4 12h16M4 18h16');
    });

    it('should show close icon when menu is open', () => {
        render(<MobileMenuButton isOpen={true} />);

        const button = screen.getByRole('button');
        const svg = button.querySelector('svg');
        const path = svg?.querySelector('path');

        expect(path).toHaveAttribute('d', 'M6 18L18 6M6 6l12 12');
    });

    it('should update aria-label based on isOpen state', () => {
        const { rerender } = render(<MobileMenuButton isOpen={false} />);

        let button = screen.getByRole('button');
        expect(button).toHaveAttribute('aria-label', 'Open mobile menu');

        rerender(<MobileMenuButton isOpen={true} />);

        button = screen.getByRole('button');
        expect(button).toHaveAttribute('aria-label', 'Close mobile menu');
    });

    it('should update aria-expanded based on isOpen state', () => {
        const { rerender } = render(<MobileMenuButton isOpen={false} />);

        let button = screen.getByRole('button');
        expect(button).toHaveAttribute('aria-expanded', 'false');

        rerender(<MobileMenuButton isOpen={true} />);

        button = screen.getByRole('button');
        expect(button).toHaveAttribute('aria-expanded', 'true');
    });

    it('should apply custom className', () => {
        const customClass = 'custom-mobile-button';
        render(<MobileMenuButton className={customClass} />);

        const button = screen.getByRole('button');
        expect(button).toHaveClass(customClass);
    });

    it('should be hidden on desktop by default', () => {
        render(<MobileMenuButton />);

        const button = screen.getByRole('button');
        expect(button).toHaveClass('md:hidden');
    });

    it('should have proper accessibility attributes', () => {
        render(<MobileMenuButton />);

        const button = screen.getByRole('button');
        expect(button).toHaveAttribute('type', 'button');
        expect(button).toHaveAttribute('aria-label');
        expect(button).toHaveAttribute('aria-expanded');
    });

    it('should have hover styles', () => {
        render(<MobileMenuButton />);

        const button = screen.getByRole('button');
        expect(button).toHaveClass('hover:text-blue-600', 'transition-colors');
    });

    it('should not crash when clicked without onClick prop', () => {
        render(<MobileMenuButton />);

        const button = screen.getByRole('button');
        expect(() => fireEvent.click(button)).not.toThrow();
    });

    it('should match snapshot when closed', () => {
        const { container } = render(<MobileMenuButton isOpen={false} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot when open', () => {
        const { container } = render(<MobileMenuButton isOpen={true} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
