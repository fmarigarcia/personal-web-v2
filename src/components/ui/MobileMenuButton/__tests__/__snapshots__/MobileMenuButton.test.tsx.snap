// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMenuButton should match snapshot when closed 1`] = `
<button
  aria-expanded="false"
  aria-label="Open mobile menu"
  class="md:hidden p-2 text-gray-700 hover:text-blue-600 transition-colors "
  type="button"
>
  <svg
    class="w-6 h-6"
    fill="none"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
  >
    <path
      d="M4 6h16M4 12h16M4 18h16"
    />
  </svg>
</button>
`;

exports[`MobileMenuButton should match snapshot when open 1`] = `
<button
  aria-expanded="true"
  aria-label="Close mobile menu"
  class="md:hidden p-2 text-gray-700 hover:text-blue-600 transition-colors "
  type="button"
>
  <svg
    class="w-6 h-6"
    fill="none"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
  >
    <path
      d="M6 18L18 6M6 6l12 12"
    />
  </svg>
</button>
`;
